src/main/java/net/kyori/ansi/ColorLevel.java

```diff
@@ -194,20 +194,25 @@ public enum ColorLevel {
 
     // TODO
     // https://github.com/termstandard/colors
-    if (COLORTERM != null && (COLORTERM.equals("truecolor") || COLORTERM.equals("24bit"))) {
-      return ColorLevel.TRUE_COLOR;
-    } else if (TERM != null && TERM.contains("truecolor")) {
-      return ColorLevel.TRUE_COLOR;
-    } else if (TERM != null && TERM.contains("256color")) {
-      return ColorLevel.INDEXED_256;
-    } else if (TERM == null) {
-      if (WT_SESSION != null) {
-        return ColorLevel.TRUE_COLOR; // Windows Terminal
-      } else if (System.console() != null && JAnsiColorLevel.isAvailable()) {
+    if (COLORTERM != null) {
+      if (COLORTERM.contains("truecolor") || COLORTERM.contains("24bit")) {
+        return ColorLevel.TRUE_COLOR;
+      }
+    } else if (TERM != null) {
+      if (TERM.contains("truecolor") || TERM.contains("-direct")) {
+        return ColorLevel.TRUE_COLOR;
+      } else if (TERM.contains("-256color")) {
+        return ColorLevel.INDEXED_256;
+      }
+    } else if (WT_SESSION != null) {
+      return ColorLevel.TRUE_COLOR; // Windows Terminal
+    } else if (System.console() != null) {
```

Comment:
just thinking, didn't this change in some JDK rev to never return null? we should maybe do a MRjar for that (as a separate project at this point)

Reply:
the docs still say it can return null at least 🤷 
https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/lang/System.html#console()
but the code there def was touched "recently" and I think you are right?
https://github.com/openjdk/jdk/blame/master/src/java.base/share/classes/java/io/Console.java#L659-L692